### ISSUE TITLE
%Y and %C don't zero-pad short centuries like %F does

### ISSUE THREAD

**Issue by petdance** [houseabsolute/DateTime.pm#83]
I'm using DateTime 1.26 but it behaves identically in 1.49.

I had a surprising formatting problem when I had a date with an incorrect year (206 instead of 2016).  The year came out as `206` instead of `0206` when formatted with %C and %Y under `strftime`.

The docs say %C is "The century number (year/100) as a 2-digit integer."

%Y is "The year as a decimal number including the century."

%F is "Equivalent to %Y-%m-%d (the ISO 8601 date format)" but that is not the case as the code below shows.

```
my $normal_year = DateTime->new(
    year => 2016,
    month => 5,
    day => 8,
);
my $short_year  = DateTime->new(
    year => 206,
    month => 5,
    day => 8,
);
my @formats = qw( %Y-%m-%d %C%y-%m-%d %F );

for my $dt ( $normal_year, $short_year ) {
    for my $fmt ( @formats ) {
        say $dt->strftime( $fmt ), ' with ', $fmt;
    }
}
```
gives this output
```
2016-05-08 with %Y-%m-%d
2016-05-08 with %C%y-%m-%d
2016-05-08 with %F
206-05-08 with %Y-%m-%d
206-05-08 with %C%y-%m-%d
0206-05-08 with %F
```

**Comment by petdance** [comment]
To be clear, I believe that %F is correct in its padding, and that %C and %Y are incorrect in not padding.

**Comment by autarch** [comment]
Actually, according to my tests with `POSIX::strftime`, the padding in `%F` is incorrect. The strftime output should always match POSIX.

**Comment by petdance** [comment]
Can't argue with POSIX.

**Comment by petdance** [comment]
The problem came about because we have a date formatted by Oracle with the CLDR pattern of `TO_CHAR( date, 'YYYY-MM-DD' )`.

**Comment by autarch** [comment]
Fix coming shortly.

**Comment by petdance** [comment]
Well don't rush on my account.  Your fix will break my workaround of using %F. :-)

**Comment by autarch** [comment]
Fixed in v1.50
